package.json

```diff
@@ -64,11 +64,10 @@
         "@changesets/cli": "^2.26.0"
     },
     "pnpm": {
-        "packageExtensions": {
-            "@docusaurus/theme-live-codeblock": {
-                "peerDependencies": {
-                    "react-live": "^2.4.1"
-                }
+        "peerDependencyRules": {
+            "allowedVersions": {
+                "react": "17",
+                "react-dom": "17"
```

Comment:
It manipulates the package-lock and adds `|| 17` to the required react version of all dependencies. 

In case of the problematic package `react-simple-code-editor` it changes the version from 

```
react: ^16.0.0
react-dom: ^16.0.0
```

to

```
react: ^16.0.0 || 17
react-dom: ^16.0.0 || 17
```

This silences the error. I don't think that this setting does anything else

Reply:
However, actual peer dependency problems with react could go unnoticed because of this setting

Reply:
And you tried overriding the peer dependencies of `react-simple-code-editor` using `packageExtensions`?